File: src/test/com/intellij/lang/jsgraphql/highlighting/GraphQLHighlightingTest.java
package com.intellij.lang.jsgraphql.highlighting;

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase;
import com.intellij.testFramework.HighlightTestInfo;

public class GraphQLHighlightingTest extends GraphQLTestCaseBase {

    @Override
    protected String getBasePath() {
        return "/highlighting";
    }

    public void testOperationDefinitions() {
        doTest();
    }

    public void testFragmentDefinition() {
        doTest();
    }

    public void testFragmentSpread() {
        doTest();
    }

    public void testTypeNameDefinitions() {
        doTest();
    }

    public void testFields() {
        doTest();
    }

    public void testFieldDefinitions() {
        doTest();
    }

    public void testInputFieldDefinitions() {
        doTest();
    }

    public void testArgumentDefinitions() {
        doTest();
    }

    public void testArguments() {
        doTest();
    }

    public void testVariables() {
        doTest();
    }

    public void testValues() {
        doTest();
    }

    public void testDirectives() {
        doTest();
    }

    public void testKeywordsAsIdentifiers() {
        doTest();
    }

    private void doTest() {
        HighlightTestInfo highlightTestInfo = myFixture.testFile(getTestName(false) + ".graphql");
        highlightTestInfo.checkSymbolNames();
        highlightTestInfo.test();
    }
}

File: tests/testSrc/com/intellij/lang/jsgraphql/highlighting/GraphQLHighlightingTest.kt
package com.intellij.lang.jsgraphql.highlighting

import com.intellij.lang.jsgraphql.GraphQLTestCaseBase
import com.intellij.openapi.progress.runBlockingCancellable

class GraphQLHighlightingTest : GraphQLTestCaseBase() {
  override fun getBasePath(): String {
    return "/highlighting"
  }

  fun testOperationDefinitions() = runBlockingCancellable {
    doTest()
  }

  fun testFragmentDefinition() = runBlockingCancellable {
    doTest()
  }

  fun testFragmentSpread() = runBlockingCancellable {
    doTest()
  }

  fun testTypeNameDefinitions() = runBlockingCancellable {
    doTest()
  }

  fun testFields() = runBlockingCancellable {
    doTest()
  }

  fun testFieldDefinitions() = runBlockingCancellable {
    doTest()
  }

  fun testInputFieldDefinitions() = runBlockingCancellable {
    doTest()
  }

  fun testArgumentDefinitions() = runBlockingCancellable {
    doTest()
  }

  fun testArguments() = runBlockingCancellable {
    doTest()
  }

  fun testVariables() = runBlockingCancellable {
    doTest()
  }

  fun testValues() = runBlockingCancellable {
    doTest()
  }

  fun testDirectives() = runBlockingCancellable {
    doTest()
  }

  fun testKeywordsAsIdentifiers() = runBlockingCancellable {
    doTest()
  }

  fun testSchemaDefinition() = runBlockingCancellable {
    doTest()
  }

  private suspend fun doTest() {
    val highlightTestInfo = myFixture.testFile(getTestName(false) + ".graphql")
    highlightTestInfo.checkSymbolNames()
    highlightTestInfo.test()
  }
}
